semgrep-core/src/parsing/Test_parsing.ml

```diff
@@ -186,8 +187,9 @@ let parsing_common ?(verbose = true) lang xs =
          let stat =
            try
              let res =
-               Parse_target.parse_and_resolve_name_use_pfff_or_treesitter lang
-                 file
+               Common.timeout_function ~verbose:false timeout_seconds (fun () ->
```

Comment:
In the code below that catch the exn, instead of PI.bad_stat file, we could also set the PI.have_timeout to true
in case of a Timeout exn.
It's probably not used by the dashboard metrics, but we may at some point.